presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/AddIntermediateAggregations.java

```diff
@@ -206,12 +214,18 @@ private PlanNode addGatheringIntermediate(AggregationNode aggregation, PlanNodeI
         for (Map.Entry<Symbol, Aggregation> entry : assignments.entrySet()) {
             // Should only have one input symbol
             Aggregation aggregation = entry.getValue();
-            checkArgument(
-                    aggregation.getArguments().size() == 1 && !aggregation.getOrderBy().isPresent() && !aggregation.getFilter().isPresent(),
-                    "Aggregation should only have one argument and should have no order by  or filter to be able to rewritten to intermediate form");
-            Symbol input = getOnlyElement(extractUnique(entry.getValue()));
+            RowExpression argument = getOnlyElement(aggregation.getArguments()
+                    .stream()
+                    .filter(AddIntermediateAggregations::isNotLambda)
```

Comment:
The original logic does not actually cover the lambda argument case (as it was added later).

Reply:
We don't need extractUnique here since we should not have anything like `a+1` or `a+b` in aggregation function in FINAL step (see `PushPartialAggregationThroughExchange`).